### PR TITLE
Adds isCancelled, isExecuted, and clone to grpc interfaces

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt [hunk 1]
```diff
@@ -576,6 +585,127 @@ class GrpcClientTest {
     }
   }
 
+  @Test
+  fun grpcCallIsCancelledWhenItShouldBe() {
```
Comment (swankjesse) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: Nit: canceled with one L

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt [hunk 2]
```diff
@@ -576,6 +585,127 @@ class GrpcClientTest {
...
+    assertThat(grpcStreamingCall.isExecuted()).isTrue()
+
+    runBlocking {
+      assertThat(responseChannel.receive()).isEqualTo(RouteNote(message = "welcome"))
+      requestChannel.close()
+      assertThat(responseChannel.receiveOrNull()).isNull()
+    }
+  }
+
+  @Test @Ignore("Timeing out")
```
Comment (swankjesse) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: Nit: timing out

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt (hunk 1 above)
Comment (oldergod) [reply wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: > Both spellings are correct; Americans favor canceled (one L), while cancelled (two Ls) is preferred in British English and other dialects. However, there is only one correct spelling of the word cancellation, no matter where you are.

Comment (swankjesse) [reply wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: I wish canceled took 2 Ls and it's was the possessive of it.

**Pull request merged**